clippy_utils/src/sugg.rs

```diff
@@ -177,11 +177,11 @@ impl<'a> Sugg<'a> {
     pub fn ast(cx: &EarlyContext<'_>, expr: &ast::Expr, default: &'a str) -> Self {
         use rustc_ast::ast::RangeLimits;
 
-        let get_whole_snippet = || {
-            if expr.span.from_expansion() {
-                snippet_with_macro_callsite(cx, expr.span, default)
+        let snippet_without_expansion = |cx, span: Span, default| {
```

Comment:
I'm not sure if we want this in clippy utils

Reply:
For now, I would keep it local to this function. Macros are not the simplest thing to deal with, and I prefer contributors to ignore them rather than trying to add suggestions for them. But that is mostly my personal preference, you can extract it if you want :upside_down_face:

Reply:
That's fair. I'll leave it as it is then